### PR TITLE
Code cleanup for linkmod.rs

### PR DESCRIPTION
**Pull request by bpglaser**
Some code cleanup to help make the code more idiomatic.

### CODE REVIEW

File: src/linkmod.rs [hunk 1]
```diff
@@ -33,11 +33,9 @@ struct Mod {
 /// Creates an embed based on the recieved mod data. {{{1
 /// Returns true if successful.
 fn make_mod_embed(modification: Mod, message: &Message) -> bool {
-    let mut tag_str = String::from("Not tagged.");
-    if let Some(ref tag) = modification.tag {
-        tag_str = tag.clone();
-    }
-    let result = message.channel_id.send_message(move |a| {
+    let tag_str = modification.tag.as_ref().map(String::as_str).unwrap_or("Not tagged.");
```
Comment (bpglaser) [inline src/linkmod.rs:36]: Instead of allocating space for a new string on the heap, one can instead just use a &str ref.

File: src/linkmod.rs [hunk 2]
```diff
@@ -175,24 +173,20 @@ fn make_request(request: &String) -> JsonValue {
...
+    message.channel_id.send_message(|a| {
         a.embed(|b| {
             b.title("Search results:")
                 .description(&serialize_search_results(&results))
                 .color(Colour::from_rgb(255, 34, 108))
                 .timestamp(message.timestamp.to_rfc3339())
         })
-    });
-    if let Err(_) = result { false } else { true }
+    }).is_ok()
```
Comment (bpglaser) [inline src/linkmod.rs:184]: Using the Result::is_ok function instead of re-implementing.

File: src/linkmod.rs [hunk 3]
```diff
@@ -175,24 +173,20 @@ fn make_request(request: &String) -> JsonValue {
...
 }
 
 /// Takes a json array as input, returns a string of the search results serialized. {{{1
 fn serialize_search_results(results: &JsonValue) -> String {
     let mut final_string = String::new();
-    for (counter, entry) in results.members().enumerate() {
-        if counter > 10 {
-            break; //Don't put more than ten entries
-        }
+    for entry in results.members().take(10) {
```
Comment (bpglaser) [inline src/linkmod.rs]: Eliminates the need for enumerating since the value was only used for the break.

**Comment by Sublimelime** [comment]
Thanks again, I really should look into methods like .take() and .map().

**Comment by bpglaser** [comment]
No problem. Happy to help.

**Pull request merged**